class/models/Omise_Models_Charge.php

```diff
@@ -30,9 +30,10 @@ public function __construct($order_id)
      */
     public function syncOmise()
     {
-        $objCharge   = OmiseWrapper::chargeRetrieve(self::getChargeId());
+        $objOmiseWrapper = new OmiseWrapper();
```

Comment:
Is it possible to inject `OmiseWrapper` into the constructor rather than initializing it direclty within the method? Does our design allow it?

Reply:
@aashishgurung 
I think It's possible.
Will do that on next monday.

Reply:
@aashishgurung 
Done.
https://github.com/omise/omise-eccube/pull/14/commits/e4c2a28eafa0de102e61a7a499a742c4b3a4d0bd

Reply:
Thank you. This is much better but I was suggesting dependency injection like following

```
public function __construct(
    $order_id,
    SC_Helper_Purchase_Ex $helperPurchase,
    OmiseWrapper $omiseWrapper
)
{
    $this->objPurchase = $helperPurchase;
    $this->arrOrder = $this->objPurchase->getOrder($order_id);
    $this->objOmiseWrapper = $omiseWrapper;
}
```

Can we do it?

Reply:
@aashishgurung 
yes, we can do that.
Will do that.

Reply:
@aashishgurung 
done
https://github.com/omise/omise-eccube/pull/14/commits/cd12fdfee851ae0110b385b8edab7421de1f35ee